### ISSUE TITLE
Getting inconsistent data loaded for files and lists of files

### ISSUE THREAD

**Issue by jsdillon** [HERA-Team/pyuvdata#299]
Here's a small snippet that reproduces the error:

```
from pyuvdata import UVData
import numpy as np
from hera_cal.data import DATA_PATH
dataFileList = [DATA_PATH + '/zen.2457698.40355.xx.HH.uvcA',
                DATA_PATH + '/zen.2457698.40355.xy.HH.uvcA',
                DATA_PATH + '/zen.2457698.40355.yx.HH.uvcA',
                DATA_PATH + '/zen.2457698.40355.yy.HH.uvcA']
uvd1 = UVData()
uvd1.read_miriad(dataFileList)
print np.sum(np.abs(uvd1.get_data(9,10,'yy')))
uvd2 = UVData()
uvd2.read_miriad(dataFileList[3])
print np.sum(np.abs(uvd2.get_data(9,10,'yy')))
```

this yields

```
0.0
275.197
```

which doesn't make sense.

**Comment by adampbeardsley** [comment]
This definitely looks like a bug in the `__add__` function. Specifically, it's having trouble reordering the polarizations (not sure why there's a rule on the order of pols, but there you go). I'll work on a fix, but in the mean time you can work around it by listing your files in this order: xx, yy, xy, yx.

**Comment by bhazelton** [comment]
So ordering of the pols is important if you want to write it to a uvfits file because the polarization axis is represented by a value at a particular pixel and then a delta from pixel to pixel along the axis and polarizations are represented as (xx, yy, xy, yx) -> (-8, -7, -6, -5). If they aren't in a sensible order, then if you write to a uvfits file you can't set a single pixel delta to define the axis. I don't recall if this is an issue for miriad or not, I suspect it's not.

**Comment by adampbeardsley** [comment]
Ok, that makes sense. At any rate, it looks like we use the same code logic to order the frequencies, so we should make sure we get it right (and test it!). I'm looking at it now, and remembering how opaque this code is. Unfortunately I'm out of the office tomorrow and Friday, so I may not come up with a solution until next week. In the mean time Josh should be able to use the workaround I mentioned above.

**Comment by adampbeardsley** [comment]
Fixed with PR #301
